src/test/enablecodecoveragetests.ts

```diff
@@ -0,0 +1,360 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+/// <reference path="./definitions/mocha.d.ts"/>
+
+import assert = require('assert');
+import path = require('path');
+import fm = require('./lib/feedback');
+import cce = require('../agent/codecoverageenabler');
+import ccec = require('../agent/commands/codecoverage.enable');
+import cm = require('../agent/common');
+import ifm = require('../agent/interfaces');
+import testifm = require('vso-node-api/interfaces/TestInterfaces');
+import tc = require('./lib/testcommand');
+import ctxm = require('../agent/context');
+import tec = require('./lib/testExecutionContext');
+import fs = require('fs');
+
+var shell = require('shelljs');
+var jobInf = require('./lib/testJobInfo');
+
+describe('CodeCoverageEnablerTests', function() {
+
+    var invalidBuildFile = path.resolve(__dirname, './javabuildfiles/test.gradle');
+    var gradleFile = path.resolve(__dirname, './javabuildfiles/build.gradle');
+    var gradleJacocoSingleModuleFile = path.resolve(__dirname, './javabuildfiles/build_jacoco_cc.gradle');
+    var summaryFile = path.resolve(__dirname, './javabuildfiles/summary.xml');
+    var reportDir = path.resolve(__dirname);
+    var testExecutionContext;
+    var includeFilter = "+:com.*.*,+:app.me*.*";
+    var excludeFilter = "-:me.*.*,-:a.b.*,-:my.com.Test";
+    var jacocoIncludeFilter = "'com/*/*/**','app/me*/*/**'";
+    var jacocoExcludeFilter = "'me/*/*/**','a/b/*/**','my/com/Test.class'"
+    var coberturaIncludeFilter = "'.*com.*..*','.*app.me*..*'"
+    var coberturaExcludeFilter = "'.*me.*..*','.*a.b..*','.*my.com.Test'"
+
+    it('codecoverage.enable : Verify code coverage tool is not valid', function(done) {
+        this.timeout(2000);
+        var properties: { [name: string]: string } = { "codecoveragetool": "", "summaryfile": gradleFile };
+        var command: cm.ITaskCommand = new tc.TestCommand(null, null, null);
+        command.properties = properties;
+        testExecutionContext = new tec.TestExecutionContext(new jobInf.TestJobInfo({}));
+
+        var ccEnableCommand = new ccec.CodeCoverageEnableCommand(testExecutionContext, command);
+        ccEnableCommand.runCommandAsync().fail(
+            function(err) {
+                var expectedMessage = "Error: No code coverage tool is provided.";
+                assert(err == expectedMessage);
+                done();
+            });
+    })
+
+    it('codecoverage.enable : Verify build tool is not valid', function(done) {
+        this.timeout(2000);
+        var properties: { [name: string]: string } = { "codecoveragetool": "jacoco", "buildtool": "" };
```

Comment:
Make sure that these tests include all the tests from windows implementation test base